bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,276 @@
+                    response = api.postRequest(ipAddress, sessionString);
+                    if (response != null && !response.isEmpty()) {
+                        success = IntesisHomeJSonDTO.getSuccess(response);
+                        if (success) {
+                            JsonElement idNode = IntesisHomeJSonDTO.getData(response).get("id");
+                            AuthenticateData auth = gson.fromJson(idNode, AuthenticateData.class);
+                            sessionId = auth.sessionID;
+                            updateStatus(ThingStatus.ONLINE);
+                            response = api.postRequest(ipAddress, contentString);
+                            success = IntesisHomeJSonDTO.getSuccess(response);
+                            if (!success) {
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                            }
+                        } else {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        }
+                    }
+                } else {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+
+            }
+        }
+    }
+
+    /**
+     * Update device status and all channels
+     */
+    public void getAllUidValues() {
+        logger.debug("Polling IntesisHome device for actuall status");
+        if (thing.getStatus() != ThingStatus.ONLINE) {
+            String sessionString = "{\"command\":\"login\",\"data\":{\"username\":\"Admin\",\"password\":\"" + password
+                    + "\"}}";
+            String response = api.postRequest(ipAddress, sessionString);
+            if (response != null && !response.isEmpty()) {
+                boolean success = IntesisHomeJSonDTO.getSuccess(response);
+                if (success) {
+                    JsonElement idNode = IntesisHomeJSonDTO.getData(response).get("id");
+                    AuthenticateData auth = gson.fromJson(idNode, AuthenticateData.class);
+                    sessionId = auth.sessionID;
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            }
+        }
+        String contentString = "{\"command\":\"getdatapointvalue\",\"data\":{\"sessionID\":\"" + sessionId
+                + "\", \"uid\":\"all\"}}";
+        String response = api.postRequest(ipAddress, contentString);
+        logger.debug("Thing {} received response {}", this.getThing().getUID(), response);
+        if (response != null && !response.isEmpty()) {
+            boolean success = IntesisHomeJSonDTO.getSuccess(response);
+            if (!success) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            } else {
+                JsonElement dpvalNode = IntesisHomeJSonDTO.getData(response).get("dpval");
+                dpval[] uid = gson.fromJson(dpvalNode, dpval[].class);
+                int value = uid[0].value;
```

Comment:
Better check if the array has the length you expect.

Reply:
Done